### PR TITLE
Remove global RzLine instance

### PR DESCRIPTION
**Pull request by XVilka**
# DO NOT SQUASH ME

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Remove global `RzLine` instance in preparation to removing global `RzCons`

**Test plan**

CI is green

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/4055

### CODE REVIEW

File: librz/cons/dietline.c [hunk 1]
```diff
@@ -57,41 +57,41 @@ static void undo_entry_free(RzLineUndoEntry *e, void *user) {
 	RZ_FREE(e->inserted_text);
 }
 
-static bool undo_reset(void) {
-	if (I.enable_vi_mode || I.hud) {
-		// Undo functionality does not yet support vi_mode.
+static bool undo_reset(RzLine *line) {
+	if (line->enable_vi_mode || line->hud) {
+		// FIXME: Undo functionality does not yet support vi_mode.
```
Comment (wargio) [inline librz/cons/dietline.c]:
```suggestion
		// FIXME: Undo functionality is not yet support vi_mode.
```

File: librz/cons/dietline.c [hunk 2]
```diff
@@ -297,121 +298,124 @@ static inline bool is_word_break_char(char ch, bool mode) {
...
+		/* This removes any trailing spaces from the input */
+	}
+	for (; i > 0 && line->buffer.data[i] != ' '; i--) {
+		/* Move cursor backwards until we hit a space character or EOL */
+		/* This deletes everything back to the previous space character */
+	}
+	if (i > 0) {
+		i++;
+	} else if (i < 0) {
+		i = 0;
```
Comment (wargio) [inline librz/cons/dietline.c:379]: this should never happen. check if coverity says is dead code

**Review by wargio — APPROVED** [review]
lgtm

File: librz/cons/dietline.c (hunk 2 above)
Comment (XVilka) [reply librz/cons/dietline.c:379]: Why? The previous two `for` cycles can make it negative.

**Pull request merged**